pkg/configurator/crd_client.go

```diff
@@ -0,0 +1,182 @@
+package configurator
+
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/openservicemesh/osm/pkg/announcements"
+	"github.com/openservicemesh/osm/pkg/apis/config/v1alpha1"
+	"github.com/openservicemesh/osm/pkg/gen/client/config/clientset/versioned"
+	informers "github.com/openservicemesh/osm/pkg/gen/client/config/informers/externalversions"
+	k8s "github.com/openservicemesh/osm/pkg/kubernetes"
+	"github.com/openservicemesh/osm/pkg/kubernetes/events"
+)
+
+const (
+	meshConfigInformerName = "MeshConfig"
+	meshConfigProviderName = "OSM"
+)
+
+func newConfiguratorWithCRDClient(meshConfigClientSet versioned.Interface, stop <-chan struct{}, osmNamespace string) (*CRDClient, error) {
+	informerFactory := informers.NewSharedInformerFactoryWithOptions(
+		meshConfigClientSet,
+		k8s.DefaultKubeEventResyncInterval,
+		informers.WithNamespace(osmNamespace),
+	)
+	informer := informerFactory.Config().V1alpha1().MeshConfigs().Informer()
+	crdClient := CRDClient{
+		informer:     informer,
+		cache:        informer.GetStore(),
+		cacheSynced:  make(chan interface{}),
+		osmNamespace: osmNamespace,
+	}
+
+	// configure listener
+	eventTypes := k8s.EventTypes{
+		Add:    announcements.MeshConfigAdded,
+		Update: announcements.MeshConfigUpdated,
+		Delete: announcements.MeshConfigDeleted,
+	}
+	informer.AddEventHandler(k8s.GetKubernetesEventHandlers(meshConfigInformerName, meshConfigProviderName, nil, eventTypes))
+
+	// start listener
+	err := crdClient.startMeshConfigListener()
+	if err != nil {
+		return &crdClient, errors.Errorf("Could not start MeshConfig CRD client: %s", err)
+	}
+
+	// start informer
+	go informer.Run(stop)
+
+	return &crdClient, err
+}
+
+// Listens to ConfigMap events and notifies dispatcher to issue config updates to the envoys based
+// on config seen on the configmap
+func (c *CRDClient) startMeshConfigListener() error {
```

Comment:
Since you are touching this:, I was working on a commit to fix a few things around this area. Can you also add a `stop` channel to this routines? also add them on the test counterparts to avoid leaking running goroutines during test env.

reference for some of it
https://github.com/openservicemesh/osm/pull/3140/files#diff-09ca33777a9661a71336ae83cf3400056ece37a5c5287200c868b685d281a6f5R107